core/src/main/java/org/testcontainers/utility/Versioning.java

```diff
@@ -0,0 +1,64 @@
+package org.testcontainers.utility;
+
+import lombok.Data;
+
+/**
+ * Represents mechanisms for versioning docker images.
+ */
+interface Versioning {
+    boolean isValid();
+
+    String getSeparator();
+
+    @Data
```

Comment:
`@Value` instead of `@Data`? Or `@EqualsAndHashcode`? Or we don't use lombok in the first plance, since we already define `toString()` and the constructor.

Reply:
`@EqualsAndHashCode` would do the trick - good suggestion.